index.js

```diff
@@ -3,89 +3,115 @@
 // # S3 storage module for Ghost blog http://ghost.org/
 var fs = require('fs');
 var path = require('path');
-var nodefn = require('when/node/function');
-var when = require('when');
-var readFile = nodefn.lift(fs.readFile);
-var unlink = nodefn.lift(fs.unlink);
-var AWS = require('aws-sdk');
+var Bluebird = require('bluebird');
+var AWS = require('aws-sdk-promise');
```

Comment:
Using this module simplifies using the AWS SDK in a promise style, so we don't have to manage those promises ourselves.